tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -637,6 +635,7 @@ TEST_F(ValidationTest, WhenDepthNotFloatThenFail) {
 }
 
 TEST_F(ValidationTest, BarrierFail) {
+  if (!m_CompilerPreservesBBNames) return;    // This test requires name preservation to succeed currently.
```

Comment:
Blocked generally means that the test could not run because of a failure setting it up.  It should be treated as a failure when we see blocked tests.

I was looking into using Skipped, but any result other than pass shows up in the exit code as non-zero (number of non-passing tests).  We will have to update our hcttest.cmd to have more sophisticated error detection based on parsing the log in order to do anything else here.